### DISCUSSION TITLE
[Simulation question]: Saving damage to nucleus as dose increases

### DISCUSSION THREAD

**Discussion by tiagoaaazevedo1** [topas-nbio/TOPAS-nBio#47] [Simulation-question] [answered]
### Simulation question

Hello everyone! I'm doing simulations where we irradiate a cell culture, each with a TsNucleus inside, with an x-ray beam based on the 6 MV TrueBeam LINAC. Then we compare the damage done at the DNA level with the results obtained experimentally. As you all know, x-ray beams deposit very little dose. So when we want to simulate the experimental setup where the cell culture were irradiated with a dose of up to 10 Gy, the simulations can take quite a while to finish. Until now, for each dose (2 Gy, 4 Gy and so on), I did a separate simulation with the following parameter:

d:Sc/MyScorer/RepeatSequenceUntilSumGreaterThan = 2 Gy 

Where every run I would increase the dose.

Is it possible to have a single run (for 10 Gy for example), but where the scorer outputs a file with the damage at regular intervals (every Gy or every time the sequence repeats)?

Thank you very much for the help! :)
Tiago A.

### TOPAS-nBio category

Scoring

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Attestation

- [X] I testify that I have fully read the bug reporting guideline beforehand.

**Comment by tmasilela** [comment, marked as answer]
Hi Tiago,

There should be a way to do this (maybe with Time Features?) but I'm not exactly sure how.

Using [Dose.txt](https://opentopas.readthedocs.io/en/latest/examples-docs/Scoring/Dose.html) as an example, adding the parameters _RepeatSequenceUntilSumGreaterThan_, _OutputAfterRun_, and _OutputAfterRunShouldAccumulate_ (see **Dose_1.txt**), partially solves your problem (Simulations repeated until a prescribed dose, however output is based on # of histories, not dose).

An alternative solution with time features (see **Dose_2.txt**) is to replace the parameter _RepeatSequenceUntilSumGreaterThan_ with _NumberOfSequentialTimes_. Neither of these solutions gets you does exactly what you want, but hopefully that helps you in the right direction.

Another thing to note (without knowing the details of your experimental setup) is that more often than not there is no need to simulate up until the experimental prescribed dose. It is more computationally efficient to run your simulation until sufficient statistics are reached, then scale up your results to the experimental dose.

Hope this helps.

All the best,
Thongchai

[Dose_1.txt](https://github.com/user-attachments/files/17577120/Dose_1.txt)
[Dose_2.txt](https://github.com/user-attachments/files/17577243/Dose_2.txt)

**Comment by datboigbenga** [comment]
Hw

> ### Simulation question
> Hello everyone! I'm doing simulations where we irradiate a cell culture, each with a TsNucleus inside, with an x-ray beam based on the 6 MV TrueBeam LINAC. Then we compare the damage done at the DNA level with the results obtained experimentally. As you all know, x-ray beams deposit very little dose. So when we want to simulate the experimental setup where the cell culture were irradiated with a dose of up to 10 Gy, the simulations can take quite a while to finish. Until now, for each dose (2 Gy, 4 Gy and so on), I did a separate simulation with the following parameter:
> 
> d:Sc/MyScorer/RepeatSequenceUntilSumGreaterThan = 2 Gy
> 
> Where every run I would increase the dose.
> 
> Is it possible to have a single run (for 10 Gy for example), but where the scorer outputs a file with the damage at regular intervals (every Gy or every time the sequence repeats)?
> 
> Thank you very much for the help! :) Tiago A.
> 
> ### TOPAS-nBio category
> Scoring
> 
> ### Code of Conduct
> * [x]  I agree to follow this project's Code of Conduct
> 
> ### Attestation
> * [x]  I testify that I have fully read the bug reporting guideline beforehand.

Hello Sir, i'm working on a related simulation . Can i get in contact with you via your email

**Comment by jschuemann** [comment]
We encourage discourse on the forum so that others can benefit from the solutions, too.
